src/stcal/resample/utils.py

```diff
@@ -0,0 +1,513 @@
+import logging
+import warnings
+
+import numpy as np
+from scipy.ndimage import median_filter
+from astropy.nddata.bitmask import (
+    bitfield_to_boolean_mask,
+    interpret_bit_flags,
+)
+from astropy import units as u
+from spherical_geometry.polygon import SphericalPolygon  # type: ignore[import-untyped]
+
+
+__all__ = [
+    "build_driz_weight",
+    "build_mask",
+    "bytes2human",
+    "compute_mean_pixel_area",
+    "get_tmeasure",
+    "is_flux_density",
+    "is_imaging_wcs",
+    "resample_range",
+]
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def resample_range(data_shape, bbox=None):
+    # Find range of input pixels to resample:
+    if len(data_shape) != 2:
+        raise ValueError("Expected 'data_shape' to be of length 2.")
+
+    if bbox is None:
+        xmin = ymin = 0
+        xmax = max(data_shape[1] - 1, 0)
+        ymax = max(data_shape[0] - 1, 0)
+
+    else:
+        if len(bbox) != 2:
+            raise ValueError("Expected bounding box to be of length 2.")
+        ((x1, x2), (y1, y2)) = bbox
+        xmin = max(0, int(x1 + 0.5))
+        ymin = max(0, int(y1 + 0.5))
+        xmax = max(xmin, min(data_shape[1] - 1, int(x2 + 0.5)))
+        ymax = max(ymin, min(data_shape[0] - 1, int(y2 + 0.5)))
+
+    return xmin, xmax, ymin, ymax
+
+
+def build_mask(dqarr, bitvalue, flag_name_map=None):
+    """Build a bit mask from an input DQ array and a bitvalue flag
+
+    In the returned bit mask, 1 is good, 0 is bad
+    """
+    bitvalue = interpret_bit_flags(bitvalue, flag_name_map=flag_name_map)
+
+    if bitvalue is None:
+        return np.ones(dqarr.shape, dtype=np.uint8)
+    return np.logical_not(np.bitwise_and(dqarr, ~bitvalue)).astype(np.uint8)
+
+
+def build_driz_weight(model, weight_type=None, good_bits=None,
+                      flag_name_map=None):
+    """ Create a weight map that is used for weighting input images when
+    they are co-added to the ouput model.
+
+    Parameters
+    ----------
+    model : dict
+        Input model: a dictionar of relevant keywords and values.
```

Comment:
What are the relevant keywords and values? To check I'm following things, "model" here is a dictionary that needs the following keys:
- "data"
- "dq"
- "var_rnoise" (if "ivm" is used)
if "exptime" is used the dictionary needs either "measurement_time" or "exposure_time" but if "measurement_time" has a value of None then "exposure_time" must also be present.

What did I get wrong there?